### PR TITLE
Should not recognize DOM elements only by 'tagName' property (closes #942)

### LINKED ISSUE #942: Uncaught TypeError: input.getAttribute is not a function on 'https://www.twitch.tv/directory'
It's reproducing in playground.

Simplified snippet to reproduce without original page:
1. Open any page via playground, 'http://example.com' for example
2. Open DevTools and type
```js
__get$({tagName:'input'}, 'autocomplete')
```

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @churkin, @LavrovArtem, @MarinaRukavitsyna

### CODE REVIEW

File: src/client/sandbox/code-instrumentation/properties/index.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ export default class PropertyAccessorsInstrumentation extends SandboxBase {
             },
 
             autocomplete: {
-                condition: domUtils.isInputElement,
+                condition: input => domUtils.isInputElement(input) && domUtils.isDomElement(input),
```
Comment (churkin) [inline src/client/sandbox/code-instrumentation/properties/index.js]: I think we should add `isDomElement` everywhere - `isInputElement`, `isHtmlElement`, `isBodyElement` etc.

File: test/client/fixtures/sandbox/code-instrumentation/getters-test.js [hunk 2]
```diff
@@ -400,3 +400,10 @@ test('should not create proxy url for invalid url (GH-778)', function () {
         strictEqual(linkVal, nativeLinkVal);
     }
 });
+
+test("Should not use 'autocomplete' getter if an object isn't DOM Element (GH-942)", function () {
```
Comment (MarinaRukavitsyna) [inline test/client/fixtures/sandbox/code-instrumentation/getters-test.js]: test("Should not use the 'autocomplete' getter if an object isn't a DOM element  (GH-942)", function () {

// NOTE: This code checks the PropertyAccessorsInstrumentation.attach(window).autocomplete condition

File: src/client/utils/dom.js [hunk 3]
```diff
@@ -347,7 +347,7 @@ export function isHammerheadAttr (attr) {
 }
 
 export function isIframeElement (el) {
-    return getTagName(el) === 'iframe';
+    return getTagName(el) === 'iframe' && isDomElement(el);
```
Comment (LavrovArtem) [inline src/client/utils/dom.js]: May be we can to move the `isDomElement` condition into the `getTagName` function?

Comment (miherlosev) [inline src/client/utils/dom.js]: Additional check `domUtils.isDomElement` needs only in code instrumentation.
If we will add it to the `domUtils.isSomeElement` then we get proxy perfomance degradation.
@churkin Are you sure that we should do it?

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
@testcafe-build-bot \retest

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by inikulin** [comment]
I guess we'll not have significant performance degradation. In most of cases we'll hit the first control path in `isDomElement` which is `instanceof` check. It's quite fast. Also, more likely `isDomElement` will be inlined by VM

File: test/client/fixtures/sandbox/code-instrumentation/getters-test.js (hunk 2 above)
Comment (MarinaRukavitsyna) [inline test/client/fixtures/sandbox/code-instrumentation/getters-test.js]: // NOTE: This code checks PropertyAccessorsInstrumentation.attach(window).autocomplete.condition

**Pull request merged**